modules/standard/Map.chpl

```diff
@@ -704,17 +722,31 @@ module Map {
     :arg lhs: The map to assign to.
     :arg rhs: The map to assign from. 
   */
-  operator map.=(ref lhs: map(?kt, ?vt, ?ps), const ref rhs: map(kt, vt, ps)) {
-
+  operator map.=(ref lhs: map(?kt, ?vt, ?), const ref rhs: map(kt, vt, ?)) {
     if !isCopyableType(kt) || !isCopyableType(vt) then
       compilerError("assigning map with non-copyable type");
 
     lhs.clear();
     for key in rhs.keys() {
-      lhs.add(key, rhs[key]);
+      lhs.add(key, rhs.getValue(key));
     }
   }
 
+  pragma "no doc"
+  operator :(x: map(?k1, ?v1, ?p1), type t: map(?k2, ?v2, ?p2)) {
+    // TODO: Allow coercion between element types? If we do then init=
+    // should also be changed accordingly.
+    if k1 != k2 then
+      compilerError('Cannot cast to map with different ',
+                    'key type: ', k2:string);
+
+    if v1 != v2 then
+      compilerError('Cannot cast to map with different ',
+                    'value type: ', v2:string);
+
+    var result: map(k1, v1, p2) = x;
```

Comment:
Can this just have `var result: t = x`? Or if you want to spell it out I'd prefer to use `k2`, `v2`, and `p2`.